File: languages/go/prime.go
package main

import (
	"fmt"
	"math"
)

func isPrime(n int) bool {
	if n <= 1 {
		return false
	}
	if n == 2 {
		return true
	}
	if n%2 == 0 {
		return false
	}

	for i := 3; i <= int(math.Sqrt(float64(n))); i += 2 {
		if n%i == 0 {
			return false
		}
	}
	return true
}

func main() {
	var number int
	fmt.Print("Enter a number to check if it's prime: ")
	fmt.Scan(&number)

	if isPrime(number) {
		fmt.Printf("%d is a prime number.\n", number)
	} else {
		fmt.Printf("%d is not a prime number.\n", number)
	}
}

File: languages/kotlin/Prime.kt
import kotlin.math.sqrt

fun isPrime(n: Int): Boolean {
    if (n <= 1) return false
    if (n == 2) return true
    if (n % 2 == 0) return false

    for (i in 3..sqrt(n.toDouble()).toInt() step 2) {
        if (n % i == 0) return false
    }
    return true
}

fun main() {
    print("Enter a number to check if it's prime: ")
    val number = readLine()?.toIntOrNull()

    if (number != null) {
        if (isPrime(number)) {
            println("$number is a prime number.")
        } else {
            println("$number is not a prime number.")
        }
    } else {
        println("Invalid input.")
    }
}
